modules/apps/foundation/contacts/contacts-web/src/main/java/com/liferay/contacts/web/notifications/ContactsCenterUserNotificationHandler.java

```diff
@@ -108,6 +102,16 @@ protected String getBody(
 				new Object[] {creatorUserName});
 		}
 
+		if ((socialRequest.getStatus() !=
+				SocialRequestConstants.STATUS_PENDING) ||
+			(socialRequest.getCreateDate() >
+				userNotificationEvent.getTimestamp())) {
```

Comment:
When a new request is sent, its [create date also gets updated](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/social/service/impl/SocialRequestLocalServiceImpl.java#L87), then a notification event is sent. So all notifications created after the request will have timestamp greater than the request's create date, and all the old notifications will have timestamp lower than the request's create date.